### PR TITLE
NOISSUE - Add Requirements File

### PR DESCRIPTION
**Pull request by rodneyosodo**
### What does this do?
Adds a requirement file to be able to install a specific version of `mkdocs` that runs well. When the version isn't set `mkdocs` fails as the latest jinja version breaks serving docs locally.

### Which issue(s) does this PR fix/relate to?
NOISSUE

### List any changes that modify/break current functionality
Introduces a requirements file to be able to install a specific version of mkdocs and jinja that works

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
No

### Notes
References https://github.com/mkdocs/mkdocs/issues/2799

### CODE REVIEW

**Review by drasko — CHANGES_REQUESTED** [review]
Why all of this? Isn't `pip` supposed to pull and install all the dependencies for `mkdocs`?

**Comment by rodneyosodo** [comment]
> Why all of this? Isn't `pip` supposed to pull and install all the dependencies for `mkdocs`?

We aren't changing from `pip`. We will still be using `pip`. Since the `jinja2` 3.1.0 release `mkdocs` does not work. If you have previous version of `jinja` it will work. Moving to a requirements files makes have the specific version of `mkdocs` and `jinja` that work.

**Review by dborovcanin — DISMISSED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+click==8.1.3
+ghp-import==2.1.0
+Jinja2==3.0.0
+Markdown==3.3.7
+MarkupSafe==2.1.2
+mergedeep==1.3.4
+mkdocs==1.4.2
```
Comment (SammyOina) [inline requirements.txt]:
```suggestion
mkdocs==1.4.2
mkdocs-material==9.1.14
```

**Review by SammyOina — CHANGES_REQUESTED** [review]

**Comment by drasko** [comment]
@rodneyosodo please update this one and address the comments if needed.

File: requirements.txt (hunk 1 above)
Comment (dborovcanin) [reply requirements.txt]: @rodneyosodo What about this suggestion?

Comment (rodneyosodo) [reply requirements.txt]: It works without `mkdocs-material`

**Review by dborovcanin — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -27,6 +20,12 @@ Doc repo can be fetched from GitHub:
 git clone git@github.com:mainflux/docs.git
 ```
 
+Install [MkDocs](https://www.mkdocs.org/#installation)
+
+```bash
+pip install -r requirements.txt
```
Comment (drasko) [inline README.md:26]: What are there requirements for? These are normally for MkDocs, and are automatically pulled by `pip`.

**Review by drasko — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (SammyOina) [reply requirements.txt]: I got an error for missing the dependency, it might not work for all environements @rodneyosodo

File: README.md (hunk 2 above)
Comment (drasko) [reply README.md:26]: @rodneyosodo ^

Comment (rodneyosodo) [reply README.md:26]: The requirements file is a lock file for the specific version of dependencies for example `go.mod` file. When `Jinja2==3.1.0` was the latest release it had problems with `mkdocs` that is why I am suggesting a requirements file that locks the `mkdocs` version with a specific version of `Jinja` that works. The other alternative will be to install a specific version of `mkdocs` for example:
```bash
pip install mkdocs==1.4.3
```

Comment (drasko) [reply README.md:26]: I would just make it sure that everything works with latest `mkdoc`, and leave to user to debug if he/she cannot figure out to install most recent version.

File: requirements.txt (hunk 1 above)
Comment (rodneyosodo) [reply requirements.txt]: What is your environment, because it works without on my side?

[asciicast](https://asciinema.org/a/593519)

**Review by SammyOina — APPROVED** [review]

**Comment by rodneyosodo** [comment]
Closing this because of https://github.com/mainflux/docs/pull/137#discussion_r1242385976

**Pull request closed without merging**